### ISSUE TITLE
Knex+Backstage: Add support for 1 database/schema due to production restriction.

### ISSUE THREAD

**Issue by begrossi** [backstage/backstage#13415]
## Feature Suggestion

Here we are starting to use Backstage as our developer portal, and we have some database role restrictions in production. It's working in UAT as a one-database/multi-schema solution (with the `pluginDivisionMode=schema` option) with postgres, but now we want to put it in production and discovered that production database user roles do not permit to create schemas with the app code (only via IAC). I will be good to put all plugin tables on the same database and the same schema with pluginId as the prefix for each table, like `catalog__locations`, `catalog__entities`, `catalog__relations`, `catalog__knex_migrations`, etc.

## Possible Implementation

The actual architecture does not permit doing this without impacting all actual and future plugin codes, but I opened an issue to Knex to create a prefix for all table names on some connection. You can see it here: https://github.com/knex/knex/issues/5315. If it's accepted by Knex, we will only need to not put the pluginId as database name and schema name on connection's URI and  add pluginId as the prefix when creating a plugin's database client. (bad thing: we will have a lot of connections to the exactly same db/schema).

## Context

Despite the solution using multi-schema is a good solution (discussed on https://github.com/backstage/backstage/issues/1598), some production environments have more restricted access to avoid app creating new schemas and databases. We want to improve this solution to use all plugins on same database and same schema on postgres.

I don't see a solution without this improvement on Knex project: https://github.com/knex/knex/issues/5315 (please, vote up it!)
Maybe, a good practice to plugins using pluginId as table's prefix always solves this problem too.

Thanks in advance,
Bruno Grossi

**Comment by freben** [comment]
We also have the ability to [_not_ try to create](https://github.com/backstage/backstage/blob/master/packages/backend-common/config.d.ts#L81) databases/schemas automatically. If you enable this, you are expected to have created the required databases/schemas beforehand, for example in some management console or by asking your DBA team for it.

Would that have worked for you? The naming patterns are quite deterministic.

**Comment by taras** [comment]
We encountered a similar issue where the platform assumed that if an application requires Postgres, then it's sufficient to provide a database with credentials to access that database. We used a database deployed as a workload instead of a platform-provided DB because we could easily create databases and schemas. It would be nice to be able to create all schemas into one existing database with prefixes for plugins.
